extensions/smallrye-rest-client/deployment/src/main/java/io/quarkus/smallrye/restclient/deployment/SmallRyeRestClientProcessor.java

```diff
@@ -229,6 +230,18 @@ public void register(RegistrationContext registrationContext) {
         RestClientBuilderImpl.SSL_ENABLED = sslNativeConfig.isEnabled();
     }
 
+    @BuildStep
+    void registerProviders(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            JaxrsProvidersToRegisterBuildItem jaxrsProvidersToRegisterBuildItem) {
+        RestClientBuilderImpl.REGISTER_BUILTIN_PROVIDERS = jaxrsProvidersToRegisterBuildItem.useBuiltIn();
+        RestClientBuilderImpl.PROVIDERS_TO_REGISTER = String.join(",", jaxrsProvidersToRegisterBuildItem.getProviders());
```

Comment:
Maybe it's a stupid question but you don't want to support the gzip providers in the REST client case?

It would probably be nice to support it too, no? I think what is done in the server part indeed also considers the client case so I would just put that code in resteasy-common. And this particular config item would be moved to the common part too.

Reply:
Wouldn't the gzip providers be added when the `@Produces` and/or `@Consumes` types indicate gzip?

Do we need those providers if they don't specify gzip?

Reply:
The gzip feature is totally orthogonal. It's about the client/server supporting gzip HTTP encoding. Typically, your JSON rest service can serve your JSON compressed as gzip and the client automatically deflates it.

That's why it's an option because basically, in the rest client case, you say: "my RESTEasy installation supports gzip, you can send me gzip encoded responses".